src/core/qgsquick/qgsquickmapcanvasmap.cpp

```diff
@@ -459,9 +462,12 @@ void QgsQuickMapCanvasMap::stopRendering()
 {
   if ( mJob )
   {
+    mMapUpdateTimer.stop();
+
     disconnect( mJob, &QgsMapRendererJob::renderingLayersFinished, this, &QgsQuickMapCanvasMap::renderJobUpdated );
     disconnect( mJob, &QgsMapRendererJob::finished, this, &QgsQuickMapCanvasMap::renderJobFinished );
 
+    connect( mJob, &QgsMapRendererJob::finished, mJob, &QObject::deleteLater );
```

Comment:
```suggestion
    if ( !mJob->isActive() )
      mJob->deleteLater();
    else
      connect( mJob, &QgsMapRendererJob::finished, mJob, &QObject::deleteLater );
```

I don't know if we ever hit that scenario, but if the job is not active we will never get the finished signal.